packages/authentication/docs/authentication-strategy.md

```diff
@@ -0,0 +1,17 @@
+### Auth strategy interface
+
+```ts
+import {Request} from '@loopback/rest';
+
+interface AuthenticationStrategy {
+  // The resolver will read the options object from metadata, call `strategy.setOptions`
+  options: object;
+  authenticate(request: Request): Promise<UserProfile | undefined>;
+  setOptions(options: object);
```

Comment:
It's better to pass in the metadata via `authenticate` method, such as:
```
authenticate(request: Request, options?: ...): Promise<UserProfile | undefined>;
```

Reply:
> Your current proposal will mandate a strategy implementation to be bound in TRANSIENT scope.

True....